thirdeye-server/src/main/java/ai/startree/thirdeye/resources/CrudResource.java

```diff
@@ -318,4 +318,18 @@ public Response deleteAll(@ApiParam(hidden = true) @Auth ThirdEyePrincipal princ
     dtoManager.findAll().forEach(this::deleteDto);
     return Response.ok().build();
   }
+
+  @GET
+  @Path("/count")
+  @Timed
+  public Response countWithPredicate(
+      @ApiParam(hidden = true) @Auth ThirdEyePrincipal principal,
+      @Context UriInfo uriInfo
+  ) {
+    final MultivaluedMap<String, String> queryParameters = uriInfo.getQueryParameters();
+    final Long count = queryParameters.size() > 0
+        ? dtoManager.count(new DaoFilterBuilder(apiToIndexMap).buildFilter(queryParameters).getPredicate())
+        : dtoManager.count();
+    return Response.ok(count).build();
```

Comment:
maybe we should introduce an object

not having an object will make it harder to evolve the API.

Reply:
Adde a `CountApi` to wrap the count value.